javascript/jsx/JavaScriptParser.g4

```diff
@@ -172,7 +173,7 @@ breakStatement
 
 returnStatement
     : Return ({this.notLineTerminator()}? expressionSequence)? eos
-    | Return '(' htmlElements ')' eos
+    | Return '(' jsxElements ')' eos
```

Comment:
Is there a reason why some string literals are unfolded, like '{', but not others, like '('? Many you unfolded, e.g., line 136 with `'=' => Assign`.

Reply:
Yes.

I've added the '{' and '=', etc symbols to the new lexer modes; which means there are now multiple '{', '=' symbols. This means we can no longer match these symbols by their symbol characters in the parser, because the parser doesn't know if '{' is OpenBrace or JsxOpeningElementOpenBrace. I made these changes based on errors from antlr. For others like '(' I haven't had to touch.